integration-cli/environment/clean.go

```diff
@@ -0,0 +1,112 @@
+package environment
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"golang.org/x/net/context"
+)
+
+type testingT interface {
+	logT
+	Fatalf(string, ...interface{})
+}
+
+type logT interface {
+	Logf(string, ...interface{})
+}
+
+// Clean cleans the environment. It will keep protected object (images, container, …)
+// and remove anything else. It's meant to run after any tests so that they don't
+// depend from each others.
```

Comment:
Some nits:

"Clean the environment, preserving protected objects"

"and removing everything else"

"depend on each other"